tests/ssl-opt.sh

```diff
@@ -4863,8 +4867,176 @@ run_test    "Record Size Limit: TLS 1.3: Client-side parsing, debug output and f
+# requires_config_enabled MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE
+# run_test    "Record Size Limit: TLS 1.3: Server complies with record size limit (513), 1 fragment" \
+#             "$P_SRV debug_level=3 force_version=tls13 response_size=256" \
+#             "$G_NEXT_CLI localhost --priority=NORMAL:-VERS-ALL:+VERS-TLS1.3 -V -d 4 --recordsize 512" \
+#             0 \
+#             -c "Preparing extension (Record Size Limit/28) for 'client hello'" \
+#             -c "Sending extension Record Size Limit/28 (2 bytes)" \
+#             -s "ClientHello: record_size_limit(28) extension received."\
+#             -s "found record_size_limit extension" \
+#             -s "RecordSizeLimit: 513 Bytes" \
+#             -s "ClientHello: record_size_limit(28) extension exists." \
+#             -s "Maximum outgoing record payload length is 511" \
+#             -s "256 bytes written in 1 fragments"
+
+# requires_gnutls_tls1_3
+# requires_gnutls_record_size_limit
+# requires_config_enabled MBEDTLS_SSL_RECORD_SIZE_LIMIT
+# requires_config_enabled MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE
+# run_test    "Record Size Limit: TLS 1.3: Server complies with record size limit (513), 2 fragments" \
+#             "$P_SRV debug_level=3 force_version=tls13 response_size=768" \
+#             "$G_NEXT_CLI localhost --priority=NORMAL:-VERS-ALL:+VERS-TLS1.3 -V -d 4 --recordsize 512" \
+#             0 \
+#             -c "Preparing extension (Record Size Limit/28) for 'client hello'" \
+#             -c "Sending extension Record Size Limit/28 (2 bytes)" \
+#             -s "ClientHello: record_size_limit(28) extension received."\
+#             -s "found record_size_limit extension" \
+#             -s "RecordSizeLimit: 513 Bytes" \
+#             -s "ClientHello: record_size_limit(28) extension exists." \
+#             -s "Maximum outgoing record payload length is 511" \
+#             -s "768 bytes written in 2 fragments"
+
+# requires_gnutls_tls1_3
+# requires_gnutls_record_size_limit
+# requires_config_enabled MBEDTLS_SSL_RECORD_SIZE_LIMIT
+# requires_config_enabled MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE
+# run_test    "Record Size Limit: TLS 1.3: Server complies with record size limit (513), 3 fragments" \
+#             "$P_SRV debug_level=3 force_version=tls13 response_size=1280" \
+#             "$G_NEXT_CLI localhost --priority=NORMAL:-VERS-ALL:+VERS-TLS1.3 -V -d 4 --recordsize 512" \
+#             0 \
+#             -c "Preparing extension (Record Size Limit/28) for 'client hello'" \
+#             -c "Sending extension Record Size Limit/28 (2 bytes)" \
+#             -s "ClientHello: record_size_limit(28) extension received."\
+#             -s "found record_size_limit extension" \
+#             -s "RecordSizeLimit: 513 Bytes" \
+#             -s "ClientHello: record_size_limit(28) extension exists." \
+#             -s "Maximum outgoing record payload length is 511" \
+#             -s "1280 bytes written in 3 fragments"
```

Comment:
For the record size limit of 513, I propose to do an handshake in PSK mode (no certificate), something like:
```
requires_gnutls_tls1_3                                                          
requires_gnutls_record_size_limit                                               
requires_all_configs_enabled MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE MBEDTLS_SSL_SRV_C MBEDTLS_DEBUG_C
requires_config_enabled MBEDTLS_SSL_RECORD_SIZE_LIMIT                           
requires_config_enabled MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED        
run_test    "Record Size Limit: TLS 1.3: Server complies with record size limit (513), 1 fragment" \
            "$P_SRV debug_level=3 force_version=tls13 tls13_kex_modes=psk \     
                    psk_list=Client_identity,6162636465666768696a6b6c6d6e6f70 \ 
                    response_size=256" \                                        
            "$G_NEXT_CLI localhost --priority=NORMAL:-VERS-ALL:+VERS-TLS1.3:+PSK --recordsize 512 \
                         --pskusername Client_identity --pskkey=6162636465666768696a6b6c6d6e6f70" \
            0 \                                                                 
            -s "RecordSizeLimit: 513 Bytes" \                                   
            -s "Maximum outgoing record payload length is 511" \                
            -s "256 bytes written in 1 fragments"                               
                                                                                
requires_gnutls_tls1_3                                                          
requires_gnutls_record_size_limit                                               
requires_all_configs_enabled MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE MBEDTLS_SSL_SRV_C MBEDTLS_DEBUG_C
requires_config_enabled MBEDTLS_SSL_RECORD_SIZE_LIMIT                           
requires_config_enabled MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED        
run_test    "Record Size Limit: TLS 1.3: Server complies with record size limit (513), 2 fragments" \
            "$P_SRV debug_level=3 force_version=tls13 tls13_kex_modes=psk \     
                    psk_list=Client_identity,6162636465666768696a6b6c6d6e6f70 \ 
                    response_size=768" \                                        
            "$G_NEXT_CLI localhost --priority=NORMAL:-VERS-ALL:+VERS-TLS1.3:+PSK --recordsize 512 \
                         --pskusername Client_identity --pskkey=6162636465666768696a6b6c6d6e6f70" \
            0 \                                                                 
            -s "RecordSizeLimit: 513 Bytes" \                                   
            -s "Maximum outgoing record payload length is 511" \                
            -s "768 bytes written in 2 fragments"                               
                                                                                
requires_gnutls_tls1_3                                                          
requires_gnutls_record_size_limit                                               
requires_all_configs_enabled MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE MBEDTLS_SSL_SRV_C MBEDTLS_DEBUG_C
requires_config_enabled MBEDTLS_SSL_RECORD_SIZE_LIMIT                           
requires_config_enabled MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED        
run_test    "Record Size Limit: TLS 1.3: Server complies with record size limit (513), 3 fragments" \
            "$P_SRV debug_level=3 force_version=tls13 tls13_kex_modes=psk \     
                    psk_list=Client_identity,6162636465666768696a6b6c6d6e6f70 \ 
                    response_size=1280" \                                       
            "$G_NEXT_CLI localhost --priority=NORMAL:-VERS-ALL:+VERS-TLS1.3:+PSK --recordsize 512 \
                         --pskusername Client_identity --pskkey=6162636465666768696a6b6c6d6e6f70" \
            0 \                                                                 
            -s "RecordSizeLimit: 513 Bytes" \                                   
            -s "Maximum outgoing record payload length is 511" \                
            -s "1280 bytes written in 3 fragments
```

Reply:
Otherwise, please align the test case dependencies of the following test cases with those of the three proposed ones.

Reply:
This comment is not addressed yet it seems. Thus for tests with certificate (sizes 1024, 2048 and 4096) the dependencies should rather be:
```
requires_gnutls_tls1_3                                                          
requires_gnutls_record_size_limit                                               
requires_all_configs_enabled MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE MBEDTLS_SSL_SRV_C MBEDTLS_DEBUG_C
requires_config_enabled MBEDTLS_SSL_RECORD_SIZE_LIMIT                           
requires_config_enabled MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED
```